tests/phpt/expanders/nested_ast_expander.phpt

```diff
@@ -0,0 +1,32 @@
+--TEST--
+Provides an AST to a custom expander
+--FILE--
+<?php
+
+$(macro) {
+    $(
+        chain(
+            chain(
+                buffer("foo")
+            ) as inner
+        ) as outer
+    )
+} >> {
+    $$(\Yay\tests\fixtures\expanders\wrap_ast_expander(
+        $$(\Yay\tests\fixtures\expanders\upper_ast_expander(
+            $$(\Yay\tests\fixtures\expanders\reverse_ast_expander(
+                $(outer)
+            ))
+        ))
+    ))
```

Comment:
Hooray! We are getting there :D

Looking at the other tests in master, we can see that expanders support stuff like:

```php
$$(some_ast_based_expander($(outer) foo $(bar) whatever arbitrary token))
```

What would/should happen in this case? Thinking about the other idea we had (expanders with different signatures) this should trigger some error like:

```
Wrong argument count for `some_ast_based_expander` on `$$(some_ast_based_expander($(outer) ...`
on line %d with context [...]
```
:thinking:

Reply:
I'm not sure this is an error state. The expander has access to the same token stream it had before (which contains the interpolated data), and the AST (which contains the original matched structure). The consumer can decide which they want to use, depending on how they expect their expander to be used.

The difference is:
- the AST is from the **match**
- the token stream is from the **replace**